controller/controller.go

```diff
@@ -570,12 +571,9 @@ func UpdateSchedulerConfig(
 	operationManager *models.OperationManager,
 ) error {
 	configYAML.EnsureDefaultValues()
-
 	timeoutSec := config.GetInt("updateTimeoutSeconds")
-	lockTimeoutMS := config.GetInt("watcher.lockTimeoutMs")
-	lockKey := GetLockKey(config.GetString("watcher.lockKey"), configYAML.Name)
-	maxVersions := config.GetInt("schedulers.versions.toKeep")
-
+	timeoutDur := time.Duration(timeoutSec) * time.Second
```

Comment:
UpdateSchedulerConfig does so much work that is incredibly hard to follow :/ However, I'm not sure if it's worth looking into that at this point.